### PR TITLE
Update `_collections_abc` for 3.13

### PR DESCRIPTION
**Pull request by max-muoto**
Add new `isdisjoint` method for 3.13.

### CODE REVIEW

**Review by hauntsaninja — COMMENTED** [review]
Interestingly for frozenset we have tight typing for the Iterable generic. That might be because it's covariant, so I wonder if we could get away with that here as well.

**Comment by max-muoto** [comment]
> Interestingly for frozenset we have tight typing for the Iterable generic. That might be because it's covariant, so I wonder if we could get away with that here as well.

Yeah - we just had `Any` for `set` so wasn't totally sure. Let me give that a try.

**Comment by max-muoto** [comment]
> Interestingly for frozenset we have tight typing for the Iterable generic. That might be because it's covariant, so I wonder if we could get away with that here as well.

Actually, I think we want `Any`, frozenset rejects this:

```python
keys = frozenset(["foo", "bar", "baz"])
keys.isdisjoint({"baz", "qux"})
keys.isdisjoint(["baz", "qux"])
# Rejects
keys.isdisjoint([1, 2])
```

**Comment by max-muoto** [comment]
> > Interestingly for frozenset we have tight typing for the Iterable generic. That might be because it's covariant, so I wonder if we could get away with that here as well.
> 
> Actually, I think we want `Any`, frozenset rejects this, which seems overly strict:
> 
> ```python
> keys = frozenset(["foo", "bar", "baz"])
> keys.isdisjoint({"baz", "qux"})
> keys.isdisjoint(["baz", "qux"])
> # Rejects
> keys.isdisjoint([1, 2])
> ```

**Review by hauntsaninja — APPROVED** [review]
Yeah, container overlaps are one of the relatively few places where typeshed has sometimes tried to get a little stricter than the usual "default to minimising false positives". I think typeshed also does this for e.g. `set.__sub__`

My instinct is to try it the stricter way and relax it if we ever get a real user complaint about it, but happy to go along with whatever others think is best

File: stdlib/_collections_abc.pyi [hunk 1]
```diff
@@ -70,6 +71,8 @@ _VT_co = TypeVar("_VT_co", covariant=True)  # Value type covariant containers.
 @final
 class dict_keys(KeysView[_KT_co], Generic[_KT_co, _VT_co]):  # undocumented
     def __eq__(self, value: object, /) -> bool: ...
+    if sys.version_info >= (3, 13):
+        def isdisjoint(self, other: Iterable[Any], /) -> bool: ...
```
Comment (JelleZijlstra) [inline stdlib/_collections_abc.pyi:75]:
```suggestion
        def isdisjoint(self, other: Iterable[_KT_co], /) -> bool: ...
```
Let's start with the strict version as @hauntsaninja suggested.

File: stdlib/_collections_abc.pyi [hunk 2]
```diff
@@ -83,6 +86,8 @@ class dict_values(ValuesView[_VT_co], Generic[_KT_co, _VT_co]):  # undocumented
 @final
 class dict_items(ItemsView[_KT_co, _VT_co]):  # undocumented
     def __eq__(self, value: object, /) -> bool: ...
+    if sys.version_info >= (3, 13):
+        def isdisjoint(self, other: Iterable[Any], /) -> bool: ...
```
Comment (JelleZijlstra) [inline stdlib/_collections_abc.pyi:90]:
```suggestion
        def isdisjoint(self, other: Iterable[tuple[_KT_co, _VT_co]], /) -> bool: ...
```

**Comment by max-muoto** [comment]
Sounds good @hauntsaninja, @JelleZijlstra!

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**